src/HeapOrdering.sol

```diff
@@ -72,68 +72,55 @@ library HeapOrdering {
     /// @dev The heap may lose its invariant about the order of the values stored.
     /// @dev Only call this function with an index within array's bounds.
     /// @param _heap The heap to modify.
-    /// @param _index The index of the account in the heap to be set.
     /// @param _account The account to set the `_index` to.
+    /// @param _index The index of the account in the heap to be set.
     function setAccount(
         HeapArray storage _heap,
-        uint256 _index,
-        Account memory _account
+        Account memory _account,
+        uint256 _index
     ) private {
         _heap.accounts[_index] = _account;
         _heap.indexOf[_account.id] = _index;
     }
 
-    /// @notice Swaps two accounts in the `_heap`.
-    /// @dev The heap may lose its invariant about the order of the values stored.
-    /// @dev Only call this function with indexOf within array's bounds.
-    /// @param _heap The heap to modify.
-    /// @param _index1 The index of the first account in the heap.
-    /// @param _index2 The index of the second account in the heap.
-    function swap(
-        HeapArray storage _heap,
-        uint256 _index1,
-        uint256 _index2
-    ) private {
-        if (_index1 == _index2) return;
-        Account memory accountOldIndex1 = _heap.accounts[_index1];
-        Account memory accountOldIndex2 = _heap.accounts[_index2];
-        setAccount(_heap, _index1, accountOldIndex2);
-        setAccount(_heap, _index2, accountOldIndex1);
-    }
-
     /// @notice Moves an account up the heap until its value is smaller than the one of its parent.
-    /// @dev This functions restores the invariant about the order of the values stored when the account at `_index` is the only one with value greater than what it should be.
+    /// @dev This functions restores the invariant about the order of the values stored when the account to move is the only one with value greater than what it should be.
     /// @param _heap The heap to modify.
+    /// @param _accountToShift The account to move.
     /// @param _index The index of the account to move.
-    function shiftUp(HeapArray storage _heap, uint256 _index) private {
-        Account memory accountToShift = _heap.accounts[_index];
-        uint256 valueToShift = accountToShift.value;
+    function shiftUp(
+        HeapArray storage _heap,
+        Account memory _accountToShift,
+        uint256 _index
+    ) private {
         Account memory parentAccount;
+        uint256 parentIndex;
+
         unchecked {
             // _index is checked to be greater than 0 before subtracting 1
             while (
                 _index > ROOT &&
-                valueToShift > (parentAccount = _heap.accounts[(_index - 1) >> 1]).value
+                _accountToShift.value > (parentAccount = _heap.accounts[parentIndex = (_index - 1) >> 1]).value
```

Comment:
I tried both, the tests are slighty in favor of not saving the variable.

However, I think it depends on the situation. As I see it, saving the variable on the stack saves gas if there are many iterations since there is a constant cost at initialization and a reduced cost at each iteration. On the other hand, doing an mload at each iteration costs more at each iteration but has no initialization cost.

In my opinion, both options are plausible. I chose this solution because it showed to be slightly better, but mostly because it was done like that in the 3-heap.